index.html

```diff
@@ -957,20 +961,42 @@ <h3>
         </h3>
         <p>
           When the <a>selection</a> is dissociated with its <a>range</a>,
-          associated with a new <a>range</a> or the associated <a>range</a>'s
+          associated with a new <a>range</a>, or the associated <a>range</a>'s
           <a>boundary point</a> is mutated either by the user or the content script,
-          the user agent must <a>queue a task</a> on the <a>user interaction task source</a>
-          to <a>fire an event</a> named <code>selectionchange</code>,
-          which does not bubble and is not cancelable,
-          at the <a>document</a> associated with the <a>selection</a>.
+          the user agent must <a>schedule a selectionchange event</a> on <a>document</a>.
         </p>
         <p>
           When an [^input^] or [^textarea^] element provide a text selection
           and its selection changes (in either extent or [=direction=]),
-          the user agent must <a>queue a task</a> on the <a>user interaction task source</a>
-          to <a>fire an event</a> named <code>selectionchange</code>,
-          which bubbles but is not cancelable, at the element.
+          the user agent must <a>schedule a selectionchange event</a> on the element.
         </p>
+        <section>
+          <h4>Scheduling <code>selectionhange</code> event</h4>
+          <p>To <dfn>schedule a selectionchange event</dfn> on a node <var>target</var>, run these steps:</p>
+          <ol>
+            <li>
+              If <var>target</var>'s <a>has scheduled selectionchange event</a> flag is true,
+              abort these steps.
+            </li>
+            <li>
+              <a>Queue a task</a> on the <a>user interaction task source</a> to
```

Comment:
Instead of queue a task, you want https://html.spec.whatwg.org/#queue-an-element-task and also pass that _target_. This way some aspects of the task get better initialized.

(We want to eventually remove "queue a task".)

Reply:
I don't think that'll work because target can be a document, which isn't an element.

Reply:
Oh yeah, I guess we can invoke https://html.spec.whatwg.org/#queue-a-global-task instead and obtain the global ourselves, but it seems there should be a helper for that. Maybe the primitive should be "queue an object task" where "object" has to be a "platform object". @domenic what do you think about that?

I also wonder if someone moves an element from one document to another, should we really have the global from the former document or should we use the element's node document's global object?